spec/integration/helpers.rb

```diff
@@ -0,0 +1,24 @@
+def apply_manifest(manifest_text)
+  # Applies the given manifest in as safe a manner as possible.
+  # Note: This will try (and fail) to create files, etc.
+  Puppet[:code] = manifest_text
+  node = Puppet::Node.indirection.find(Facter.value(:fqdn))
+  catalog = Puppet::Resource::Catalog.indirection.find(
+    node.name, :use_node => node)
+  catalog = catalog.to_ral
```

Comment:
Is there any way we could avoid pulling in this code that we don't understand? Maybe `puppet-catalog-test` can do some of this stuff?